frontend/app/src/composables/accounts/blockie.ts

```diff
@@ -0,0 +1,175 @@
+  const width = size; // Only support square icons for now
+  const height = size;
+
+  const dataWidth = Math.ceil(width / 2);
+  const mirrorWidth = width - dataWidth;
+
+  const data = [];
+  for (let y = 0; y < height; y++) {
+    let row = [];
+    for (let x = 0; x < dataWidth; x++) {
+      // this makes foreground and background color to have a 43% (1/2.3) probability
+      // spot color has 13% chance
+      row[x] = Math.floor(rand() * 2.3);
+    }
+    const r = row.slice(0, mirrorWidth);
+    r.reverse();
+    row = row.concat(r);
+
+    for (const element of row) {
+      data.push(element);
+    }
+  }
+
+  return data;
+}
+
+function buildOpts(opts: Partial<BlockieOptions>): BlockieOptions {
+  const newOpts: Partial<BlockieOptions> = {};
+
+  newOpts.seed = opts.seed || Math.floor(Math.random() * 10 ** 16).toString(16);
+
+  seedRand(newOpts.seed);
+
+  // eslint-disable-next-line unicorn/explicit-length-check
+  newOpts.size = opts.size || 8;
+  newOpts.scale = opts.scale || 4;
+  newOpts.color = opts.color || createColor();
+  newOpts.bgcolor = opts.bgcolor || createColor();
+  newOpts.spotColor = opts.spotColor || createColor();
+
+  return newOpts as BlockieOptions;
+}
+
+export function renderIcon(
+  opts: Partial<BlockieOptions>,
+  canvas: HTMLCanvasElement
+) {
+  const newOpts: BlockieOptions = buildOpts(opts || {});
+  const imageData = createImageData(newOpts.size);
+  const width = Math.sqrt(imageData.length);
+
+  canvas.width = canvas.height = newOpts.size * newOpts.scale;
+
+  const cc = canvas.getContext('2d') as CanvasRenderingContext2D;
+  if (cc) {
+    cc.fillStyle = newOpts.bgcolor;
+    cc.fillRect(0, 0, canvas.width, canvas.height);
+    cc.fillStyle = newOpts.color;
+
+    for (const [i, imageDatum] of imageData.entries()) {
+      // if data is 0, leave the background
+      if (imageDatum) {
+        const row = Math.floor(i / width);
+        const col = i % width;
+
+        // if data is 2, choose spot color, if 1 choose foreground
+        cc.fillStyle = imageDatum === 1 ? newOpts.color : newOpts.spotColor;
+
+        cc.fillRect(
+          col * newOpts.scale,
+          row * newOpts.scale,
+          newOpts.scale,
+          newOpts.scale
+        );
+      }
+    }
+  }
+
+  return canvas;
+}
+
+const createIcon = (opts: Partial<BlockieOptions>) => {
+  const canvas = document.createElement('canvas');
+
+  renderIcon(opts, canvas);
+
+  return canvas;
+};
+
+export const useBlockie = createSharedComposable(() => {
+  const storedBlockie: Ref<Record<string, string>> = ref({});
+
+  const getBlockie = (address: string | null = '') => {
+    if (!address) return '';
+    const stored = { ...get(storedBlockie) };
+
+    const formatted = address.toLowerCase();
+
+    if (!stored[formatted]) {
+      const canvas = createIcon({
+        seed: formatted
+      });
+
+      stored[formatted] = canvas.toDataURL('image/jpeg');
+      set(storedBlockie, stored);
+    }
+
+    return stored[formatted];
```

Comment:
The first thing we need here, is a limit and eviction.
Similar to what we do for the asset cache.

Is storing the `canvas.toDataUrl()` more efficient than storing the base64 itself? 

If I remember the performance hit is during generation

Reply:
`canvas.toDataUrl()` will resulting the `base64` string

Reply:
Also, another important thing, can you also add some unit tests for the cache/eviction, we can just mock the actual blockie creation.

Reply:
> `canvas.toDataUrl()` will resulting the `base64` string. This is how the image is created.

OK thanks for clarifying, never looked at the library internals :)

Reply:
It would be preferable to have a limit and an eviction strategy for the cached blockies, similar to how we cache the asset information in the frontend.

My original comment was to add a similar mechanism here and add some test for that.

We can limit it to 100 cached entries and evict the least recently used items when we hit a limit.

Reply:
The limit and eviction strategy have added